ST_environs.c

```diff
@@ -191,73 +234,73 @@ static void _make_disturbance( void) {
 
 /*------------------------------------------------------*/
   /* Can't have simultaneous disturbances*/
-  if (Plot.disturbance != NoDisturb) {
-    switch( Plot.disturbance) {
+  if (Plot->disturbance != NoDisturb) {
+    switch( Plot->disturbance) {
       case FecalPat:
-           if (Plot.pat_removed) {
-             Plot.disturbed = 0;
-             Plot.pat_removed = FALSE;
-             Plot.disturbance = NoDisturb;
+           if (Plot->pat_removed) {
+             Plot->disturbed = 0;
+             Plot->pat_removed = FALSE;
+             Plot->disturbance = NoDisturb;
            } else {
-             pc = Globals.pat.recol[Slope] * Plot.disturbed
-                  + Globals.pat.recol[Intcpt];
+             pc = Globals->pat.recol[Slope] * Plot->disturbed
+                  + Globals->pat.recol[Intcpt];
              if (RandUni(&environs_rng) <= pc) {
-               Plot.pat_removed = TRUE;
+               Plot->pat_removed = TRUE;
                /* slight effects for one year*/
-               Plot.disturbed = 1;
+               Plot->disturbed = 1;
              } else {
-               Plot.pat_removed = FALSE;
-               Plot.disturbed++;
+               Plot->pat_removed = FALSE;
+               Plot->disturbed++;
              }
            }
            break;
       case NoDisturb: // Does nothing but prevent a compiler warning
       case LastDisturb:
       default:
-           Plot.disturbed = (Plot.disturbed) ? Plot.disturbed -1 : 0;
+           Plot->disturbed = (Plot->disturbed) ? Plot->disturbed -1 : 0;
```

Comment:
You don't need a break here since this is the last case, but it is a good idea to have one here anyways.

Reply:
Resolved by 9b99edd. I also changed `Plot->disturbed -1` to `(Plot->disturbed - 1)` for clarity.